stakingModel/contracts/UBIScheme.sol

```diff
@@ -151,29 +168,62 @@ contract UBIScheme is AbstractUBI {
      * has not claimed for `maxInactiveDays` is an inactive user.
      * @param account to check
      */
-    function isActiveUser(address account) public view requireActive returns (bool)
-    {
+    function isActiveUser(address account) public view requireActive returns (bool) {
         uint256 lastClaimed = lastClaimed[account];
-        if(isRegistered(account)) {
+        if (isRegistered(account)) {
             uint256 lastClaimedDay = (lastClaimed.sub(periodStart)) / 1 days;
-            if (currentDay.sub(lastClaimedDay) < maxInactiveDays) { // active user
+            if (currentDay.sub(lastClaimedDay) < maxInactiveDays) {
+                // active user
                 return true;
             }
         }
         return false;
     }
 
+    /* @dev Transfers `amount` dao tokens to `account`. updates stats
+     * and emits an event in case of claimed.
+     * @param account the account which recieves the funds
+     * @param amount the amount to transfer
+     * @param isClaimed true for claimed
+     */
+    function _transferTokens(address account, uint256 amount, bool isClaimed)
+        private
+        requireActive
+    {
+        Day storage day = claimDay[currentDay];
+        day.amountOfClaimers = day.amountOfClaimers.add(1);
+        day.claimAmount = day.claimAmount.add(amount);
+        GoodDollar token = GoodDollar(address(avatar.nativeToken()));
+        token.transfer(account, amount);
+        if(isClaimed) {
+            emit UBIClaimed(account, amount);
+        }
+    }
+
+    /* @dev Checks amount address is eligible to claim for. regardless if they have been
+     * whitelisted or not. In case the user is active, then the current day must be equal
+     * to the actual day, i.e. claim or fish has already been executed today.
+     * @return The amount of GoodDollar the address can claim.
+     */
+    function checkEntitlement() public view requireActive returns (uint256) {
```

Comment:
if user already claimed today entitlement is 0 (see current implementation)

Reply:
updated
3342c9c0a5e4ba58dae711a4431d0da78fc5094e